### PR TITLE
Carina app

### PR DESCRIPTION
**Pull request by Chulei799**

### CODE REVIEW

**Comment by Chulei799** [comment]
Test comment

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/ChartsPage.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Override
+    public void swipeToHiloDiagram() {
+        swipe(hiloChart, Direction.UP, 10, 500);
+    }
+
+    @Override
+    public void openLeftMenu() {
+        toolBar.openLeftMenu();
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/ChartsPage.java:86]: @Chulei799 avoid using method in method, if you can click on an element, click on the element itself, so it's easier to support the code

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/LoginPage.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    @Override
+    public boolean isSignUpBtnPresent() {
+        return signUpBtn.isElementPresent(FIVE_SECONDS);
+    }
+
+    @Override
+    public WelcomePageBase clickBackBtn() {
+        backBtn.click(FIVE_SECONDS);
+        return new WelcomePage(driver);
```
[the remark this answers is not in the document]
Comment (Chulei799) [reply src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/LoginPage.java:82]: Done

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    @Override
+    public boolean isProgressBarPresent(int n) {
+        switch (n) {
+            case 1:
+                return firstProgressBar.isElementPresent(FIVE_SECONDS);
+            case 2:
+                return secondProgressBar.isElementPresent(FIVE_SECONDS);
+            default:
+                return false;
+        }
```
[the remark this answers is not in the document]
Comment (Chulei799) [reply src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java:127]: Done

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/RightMenuPage.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                return installationGuide.isElementPresent(THREE_SECONDS);
+            case READ_ON_GITHUB:
+                return readOnGitHub.isElementPresent(THREE_SECONDS);
+            case DONATE:
+                return donate.isElementPresent(THREE_SECONDS);
+            case CONTACT_US:
+                return contactUs.isElementPresent(THREE_SECONDS);
+        }
+        Assert.fail("Right menu | No such component!");
+        return false;
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/RightMenuPage.java:60]: names of elements

Comment (YanaDorosh) [reply src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/RightMenuPage.java:60]: you should use flexible xPath 
    @FindBy(xpath = "//*[@content-desc='%s']/android.widget.TextView")
     private ExtendedWebElement itemByName;
     and method 
@Override
     public boolean isMenuElementPresent(RMComponent component) {
           return itemByName.format(component.getName()).isElemetPresent();
}

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    @Override
+    public boolean isViewFieldPresent(String fieldName) {
+        return viewField.format(fieldName).isElementPresent(THREE_SECONDS);
+    }
+
+    @Override
+    public boolean isTextFieldPresent() {
+        return swipe(textField, Direction.VERTICAL, 20, 500);
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java:85]: use here constants TWENTY_SWIPE_TIME, SWIPE_QUICK_DURATION

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    @Override
+    public boolean isEmailFieldPresent() {
+        return swipe(emailField, Direction.VERTICAL, 20, 500);
+    }
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/android/UIElementsPage.java:91]: how many time do you really need to swipe? and don't use Direction.VERTICAL if you don't need it

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        return viewPager.isElementPresent(FIVE_SECONDS);
+    }
+
+    public boolean isPagerDotPresent(int n) {
+        if(n < pagerDots.size()) {
+            return pagerDots.get(n).isElementPresent(FIVE_SECONDS);
+        } else {
+            return false;
+        }
+    }
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java:34]: Assert.fail ()

File: src/test/java/com/qaprosoft/carina/chulei/mobile/android/tests/AndroidLoginPageTest.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        softAssert.assertTrue(loginPage.isPrivacyPolicyCheckBoxChecked(), "Privacy Policy box not checked!");
+        loginPage.checkPrivacyPolicyCheckBox(false);
+        softAssert.assertFalse(loginPage.isPrivacyPolicyCheckBoxChecked(), "Privacy Policy box checked, but should not!");
+
+        //Scenario 1: Try login without enter user name
+        loginPage.setPasswordField(USER_PASS);
+        loginPage.selectSex(MALE);
+        loginPage.checkPrivacyPolicyCheckBox(true);
+        softAssert.assertFalse(loginPage.isSignUpBtnActive(), "Sign Up button should not be available," +
+                                                                    " if user name not entered!");
```
Comment (YanaDorosh) [inline src/test/java/com/qaprosoft/carina/chulei/mobile/android/tests/AndroidLoginPageTest.java:110]: formater

File: src/test/java/com/qaprosoft/carina/chulei/mobile/android/tests/AndroidMapPageTest.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        MapPageBase mapPage = (MapPageBase) leftMenu.openPageFromLeftMenu(LMComponent.MAP);
+        Assert.assertTrue(mapPage.isPageOpened(), "Map page isn't opened!");
+        softAssert.assertTrue(mapPage.isImageViewPresent(), "Image view isn't present!");
+        softAssert.assertTrue(mapPage.isMapPresent(), "Map isn't present!");
+        softAssert.assertTrue(mapPage.isGoogleImagePresent(), "Google image isn't present!");
+        softAssert.assertTrue(mapPage.isZoomPresent(ZOOM_IN), "Zoom in isn't present!");
+        softAssert.assertTrue(mapPage.isZoomPresent(ZOOM_OUT), "Zoom out isn't present!");
+        mapPage.clickZoom(ZOOM_IN);
+        mapPage.clickZoom(ZOOM_OUT);
+
```
Comment (YanaDorosh) [inline src/test/java/com/qaprosoft/carina/chulei/mobile/android/tests/AndroidMapPageTest.java:46]: without any asserts this clicking is useless

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java (hunk 7 above)
Comment (YanaDorosh) [reply src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java:34]: n? use understandable names

File: src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+    public boolean isAllPagerDotsPresent() {
+        return isPagerDotPresent(0) && isPagerDotPresent(1) && isPagerDotPresent(2) && isPagerDotPresent(3)
+                && isPagerDotPresent(4);
+    }
+}
```
Comment (YanaDorosh) [inline src/main/java/com/qaprosoft/carina/chulei/mobile/gui/pages/components/ImageView.java:40]: you can check this in the test method using 
int dotAmount = 4;
while(dotAmount-- > 0){
softAssert.True(pageName.isPagerDotPresent(dotAmount), "Assert text");
}

**Pull request merged**